src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -91877,5 +91877,123 @@ class Derived<T> : Base<T> where T : class
             Assert.False(dGoo.Parameters[0].Type.IsNullableType());
             Assert.True(dGoo.Parameters[0].TypeWithAnnotations.NullableAnnotation == NullableAnnotation.Annotated);
         }
+
+        [Fact, WorkItem(34299, "https://github.com/dotnet/roslyn/issues/34299")]
+        public void CheckLambdaInArrayInitializer()
+        {
+            var source =
+@"using System;
+
+class C
+{
+    static void G(object? o, string s)
+    {
+        Func<object> f = () => o; // 1
+        _ = f /*T:System.Func<object!>!*/;
+        var fa3 = new[] {
+            f,
+            () => o, // 2
+            () => {
+                    s = null; // 3
+                    return null; // 4
+                },
+            };
+        _ = fa3 /*T:System.Func<object!>![]!*/;
+        fa3[0]().ToString();
+    }
+}
```

Comment:
Could you also add tests for lambdas with multiple returns and ternaries?

Reply:
Lambdas with multiple returns are tested below in `CheckLambdaInLambdaInference`

I tested ternaries and found that the behavior was wrong and filed [https://github.com/dotnet/roslyn/issues/35029](https://github.com/dotnet/roslyn/issues/35029)

---
In reply to: [276122618](https://github.com/dotnet/roslyn/pull/35030#discussion_r276122618) [](ancestors = 276122618)